wgsl/index.bs

```diff
@@ -1366,86 +1375,360 @@ Therefore, a data value must not be placed in the padding at the end of a struct
+<div class='example wgsl' heading='Using a pointer as a formal parameter'>
+  <xmp highlight='rust'>
+    fn add_one(x: ptr<function,i32>) {
+      // Update the locations for 'x' to contain the next higher integer value,
+      // (or to wrap around to the largest negative i32 value).
+      // On the left-hand side, unary '*' converts the pointer to a reference that
+      // can then be assigned to.
+      // On the right-hand side:
+      //    - Unary '*' converts the pointer to a reference
+      //    - The only matching type rule is for addition (+) and requires '*x' to
+      //      have type i32, which is the store type for '*x'.  So the Load Rule
+      //      applies and '*x' evaluates to the value stored in the memory for '*x'
+      //      at the time of evaluation, which is the i32 value for 0.
+      //    - Add 1 to 0, to produce a final value of 1 for the right-hand side.
+      // Store 1 into the memory for '*x'.
+      *x = *x + 1u;
+    }
 
-A pointer value *P* supports the following operations:
+    [[stage(compute)]]
+    fn main() {
+      var i: i32 = 0;
+
+      // Modify the contents of 'i' so it will contain 1.
+      // Use unary '&' to get a pointer value for 'i'.
+      // This is a clear signal that the called function has access to the storage
+      // for 'i', and may modify it.
+      add_one(&i);
+      const one: i32 = i;  // 'one' has value 1.
+    }
+  </xmp>
+</div>
 
-<table class='data'>
-  <tr><td>P.Write(V)<td>Place a value V into the referenced storage.
-               V’s type must match P’s pointee type.
-  <tr><td>P.Read()<td>An evaluation yielding the value currently in the P’s
-             referenced storage.  The result type is P's pointee type.
-  <tr><td>P.Subaccess(K)<td>Valid for pointers with a composite pointee type where
-                   *K* must evaluate to an integer between 0 and one
-                   less than the number of components in *P*’s pointee type.
-                   The subaccess evaluation yields a pointer to the storage for
-                   the K’th component within P’s referenced storage,
-                   using zero-based indexing. If P's [=storage class=] is SC, and
-                   the K'th member of P's pointee type is of type T, then
-                   the result type is `ptr<SC,T>`.
-</table>
+### Forming reference and pointer values ### {#forming-a-reference-value}
+
+A reference value is formed in one of the following ways:
+
+* The identifer [=resolves|resolving=] to an [=in scope|in-scope=] variable *v* denotes the reference value for *v*'s storage.
+    * The resolved variable is the [=originating variable=] for the reference.
+* Use the unary `*` operator on a pointer.
+        See #indirection-expr.
+    * The originating variable of the result is defined as the originating variable of the pointer.
+* Use a <dfn noexport>composite value reference sub-expression</dfn> expression.  
```

Comment:
"sub-expression expression", should this just be "sub-expression"?

Reply:
Even then it's a mouthful,  maybe it should be 

> composite sub-component reference expression

Reply:
Or better: composite component reference expression